### ISSUE TITLE
type stability of return from async.map (0.1.22)

### ISSUE THREAD

**Issue by alexjeffburke** [caolan/async#197]
Hi,

I've noticed rather annoying behaviour with async.map and the types it returns - namely it seems that if async.map ever encounters a situation where it is executed on an array that only has one value, the callback(err, results) will be triggered but results will NOT be an array.

What this means you can't easily loop calling async.map multiple times and then combine the results as the calling code becomes unnecessarily complicated as it must check whether the the output was an array.

In summary async.map behaves differently with an array of 1 element than it does with an array containing more than one and I'm not sure this is intended behaviour.

Let me know your thoughts & I'd be happy to look into writing a patch for this if need be.

Thanks, Alex.
